packages/engine/lib/execution/src/runner/javascript/mod.rs

```diff
@@ -850,7 +850,10 @@ impl<'s> ThreadLocalRunner<'s> {
 
     pub fn new(scope: &mut v8::HandleScope<'s>, init: &ExperimentInitRunnerMsg) -> Result<Self> {
         let embedded = Embedded::import_common_js_files(scope)?;
-        let datasets = Self::load_datasets(scope, &init.shared_context)?;
+        let datasets = {
+            let upgraded = init.shared_context.upgrade().unwrap();
```

Comment:
Should we have an `expect` here instead?

Reply:
I mean I'm not against us having `unwrap` if it 100% will not happen. But otherwise I think we tend to say use `expect` by default